### PR TITLE
feat(document_db): add resources

### PR DESCRIPTION
**Pull request by Codelax**

### CODE REVIEW

File: scaleway/resource_document_db_read_replica_test.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+			{
+				Config: `
+				resource "scaleway_document_db_instance" "instance" {
+				  name              = "test-document_db-read-replica-basic"
+				  node_type         = "docdb-play2-pico"
+				  engine            = "FerretDB-1"
+				  is_ha_cluster     = false
+				  user_name         = "my_initial_user"
+				  password          = "thiZ_is_v&ry_s3cret"
+				  tags              = ["terraform-test", "scaleway_document_db_instance", "minimal"]
```
Comment (remyleone) [inline scaleway/resource_document_db_read_replica_test.go:119]: Remove it or test it

File: scaleway/resource_document_db_read_replica_test.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+				  volume_size_in_gb = 20
+				  telemetry_enabled = false
```
Comment (remyleone) [inline scaleway/resource_document_db_read_replica_test.go:121]: Remove it or test it

File: scaleway/helpers_document_db.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	documentdb "github.com/scaleway/scaleway-sdk-go/api/documentdb/v1beta1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+const (
+	telemetryReporting                 = "telemetry_reporting"
```
[the remark this answers is not in the document]
Comment (Monitob) [reply scaleway/helpers_document_db.go:15]: Today is the only Scaleway service exposing telemetry in order to support FerretDB OSS. Ok to rename the variable.

File: scaleway/resource_document_db_instance.go [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+
+	_ = setInitSettings(d, instance.InitSettings)
+
+	return nil
+}
+
+func setInitSettings(d *schema.ResourceData, settings []*documentdb.InstanceSetting) error {
+	for _, s := range settings {
+		if s.Name == telemetryReporting {
+			_ = d.Set("telemetry_enabled", s.Value)
```
[the remark this answers is not in the document]
Comment (Monitob) [reply scaleway/resource_document_db_instance.go]: This is only the name on our schema and the linter accepts only a string this way

**Review by remyleone — APPROVED** [review]

**Pull request merged**